packages/storage/src/indexed-db/index.ts

```diff
@@ -802,4 +795,44 @@ export class IndexedDb implements IndexedDbInterface {
       output: Value.fromJson(result.output),
     };
   }
+
+  async fetchStakingTokenId(): Promise<AssetId> {
+    const registryClient = new ChainRegistryClient();
+    const registry = registryClient.get(this.chainId);
+    const stakingTokenMetadata = registry.getMetadata(registry.stakingAssetId);
+    const stakingToken = getAssetId(stakingTokenMetadata);
+
+    return stakingToken;
+  }
+
+  async hasStakingAssetBalance(assetId: AssetId): Promise<boolean> {
+    const spendableUMNotes = await this.db.getAllFromIndex(
+      'SPENDABLE_NOTES',
+      'assetId',
+      uint8ArrayToBase64(assetId.inner),
+    );
+
+    // Iterate over the spendable UM notes, and acrue balance for unspent notesz
+    let stakingTokenBalance = new Amount();
+    for (const note of spendableUMNotes) {
+      let umNote = note as unknown as SpendableNoteRecord;
```

Comment:
```suggestion
      const umNote = SpendableNoteRecord.fromJson(note);
```

(right?)

Reply:
yeah exactly, made the same change locally!